### ISSUE TITLE
Can't open document with webdav(s) url (dav:// or davs://) in Libreoffice 5.1

### ISSUE THREAD

**Issue by CesarCapillas** [CesarCapillas/alfresco-share-online-edition-addon#6]
Apparently dav:// and davs:// urls are not working in Libreoffice 5.1.

More information:

https://bugs.documentfoundation.org/show_bug.cgi?id=93327

**Comment by CesarCapillas** [comment]
From Giussepe Castagno (Libreoffice):

The current status is:

AOO 4.x, LO 4.x: dav://, davs://, vnd.sun.star.webdav:// all link to the 
WebDAV provider (e.g. the client).

In LO 5.0 specifically: on Windows dav(s):// no longer work, in Linux 
they may still work, but it will depend from the build configuration.
vnd.sun.star.webdav:// works as before both for Linux and Windows, 
vnd.sun.star.webdavs:// it's not implemented.

In LO 5.1 vnd.sun.star.webdav:// vnd.sun.star.webdavs:// works 
specifically for WebDAV LO client and are reserved for that use.

Summing up:
- for AOO 4.x LO 4.x use dav(s)://;
- LO 5.0 is not supported well, as you can see from above;
- LO 5.1 up use vnd.sun.star.webdav(s)://.

https://wiki.documentfoundation.org/ReleaseNotes/5.1#WebDav_Improvements

**Comment by CesarCapillas** [comment]
Some checks and tests done:

- Tested the current version of the plugin (version 0.2) with LO 5.1 and it does not work for davs://  (Alfresco 5.0.c CE) 
- Tested the new url scheme vnd.sun.star.webdav:// with Libreoffice 5.1 (in Windows) and it seems to work (Alfresco 5.0.2.5 EE). My Libreoffice 4.2.8.2 with vnd.sun.star.webdav:// also is working.

TODO: Check vnd.sun.star.webdavs://

--C.

**Comment by CesarCapillas** [comment]
Commiting and publishing a pre-release for Libreoffice:

https://github.com/CesarCapillas/alfresco-share-online-edition-addon/releases/tag/0.3

**Comment by gjongerh** [comment]
Testing with Alfresco 5.1 and LibreOffice 5.1.0.3 on a iMac. Unfortunately it isn't working
After editing Info.plist the browser recognize LibreOffice as the program to use, but LibreOffice doesn't open the document(s) 
    <key>CFBundleURLTypes</key>
    <array>
        <dict>
            <key>CFBundleURLName</key>
            <string>LibreOffice CMIS</string>
            <key>CFBundleURLSchemes</key>
            <array>
                <string>vnd.libreoffice.cmis</string>
                <string>vnd.sun.star.webdavs</string>
            </array>
        </dict>
    </array>

Does anybody know if this is a bug?

**Comment by gjongerh** [comment]
Extra info after giving the following command LibreOffice does open the document, but stil not from Alfresco (in Chrome)
/Applications/LibreOffice.app/Contents/MacOS/soffice vnd.sun.star.webdavs://apollo.virtorg.nl/alfresco/webdav/virtorg/documentLibrary/01_management/beleid/2016/VirtOrg2016-MarketingPlan.odt

When typen in the browser
vnd.sun.star.webdavs://apollo.virtorg.nl/alfresco/webdav/virtorg/documentLibrary/01_management/beleid/2016/VirtOrg2016-MarketingPlan.odt
It does open LibreOffice but not the document

Don't know where to start debugging or watching for logging

**Comment by CesarCapillas** [comment]
Hi:

In principle, we did not test this plugin in MacOSX, cause the way of registering protocols in MacOSX is slightly different. Our conclusions is that in MacOSX, we need not only to register protocols as you did in Info,plist, also it is needed that LO binary were able to accept parameters (in this case de dav url).

Some notes:

- May you check if Tools > Options > Libreoffice > General > Use Libreoffice dialogs is enabled ?. If not check it.

- Also you can try with open command in a MacOSX terminal to check if it is correctly registered the protocol.

open vnd.sun.star.webdavs://apollo.virtorg.nl/alfresco/webdav/virtorg/documentLibrary/01_management/beleid/2016/VirtOrg2016-MarketingPlan.odt

- Finally, try this url in Libreoffice File dialog. If this last fails, maybe it is not possible for Libreoffice to open url-schemed files in MacOSX. 

I don't have a MacOSX computer now for testing, but I could check it next week.

Regards.

--C.

**Comment by gjongerh** [comment]
Hi, thank you very much for the info.
Yes I did all this. LibreOffice is opening but not the document
When configuring LibreOffice for remote files File->Open Remote File it is working correctly.

So I think it is a bug in LibreOffice

It would be nice to know I you have the same problem on a MAC. Then I know that it is not a problem on all my MAC computers.

Regards

**Comment by CesarCapillas** [comment]
Tests with Libreoffice 5.1.5.2 (using Libreoffice dialogs) in MacOSX 10.11.6

With the next config in /Applications/Libreoffice.app/Config/Info.plist:

```
    <key>CFBundleURLTypes</key>
    <array>
        <dict>
            <key>CFBundleURLName</key>
            <string>LibreOffice CMIS</string>
            <key>CFBundleURLSchemes</key>
            <array>
                <string>vnd.libreoffice.cmis</string>
            </array>
        </dict>
        <dict>
            <key>CFBundleURLName</key>
            <string>LibreOffice Alfresco</string>
            <key>CFBundleURLSchemes</key>
            <array>
                <string>vnd.sun.star.webdavs</string>
                <string>vnd.sun.star.webdav</string>
                <string>dav</string>
                <string>davs</string>
            </array>
        </dict>
    </array>

```

**Test 1:**

I clicked on the link in Alfresco Share, and it opens Libreoffice but not the file (as you noticed).


**Test 2:**  (using terminal command open - same behaviour)

`macmini:Downloads cesar$ open vnd.sun.star.webdavs://services.zylk.net/alfresco/webdav/Sitios/ecm/documentLibrary/Demo/EdicionOnlineLO/ECMForm_ES.odt`

It opens LibreOffice but not the remote file.


**Test 3:** (using terminal command open but with a local file)

`macmini:Downloads cesar$ open ECMForm_ES.odt`

But it does it correctly with a local file


**Test 4:** If I put the vnd.sun.star.webdavs:// url file in File Dialog in Libreoffice, it is working fine. It opens the remote file asking for Alfresco authentication

**Test 5:**  If I run this, also it is opening the remote file

```
$ cd /Applications/LibreOffice.app/Contents/MacOS
$ ./soffice vnd.sun.star.webdavs://services.zylk.net/alfresco/webdav/Sitios/ecm/documentLibrary/Demo/EdicionOnlineLO/ECMForm_ES.odt
```

But not via open -a command

Maybe the upper config for url scheme is not enough for the operating system, but it seems Libreoffice is not dealing fine with URL parameter. 

Kind Regards.

--C.

**Comment by gjongerh** [comment]
Hi, I can reproduce the same result. 

Yes it seems that LibreOffice is dealing well with the URL parameter. The OS is starting LibreOffice and is able to parse a parameter. This is proofed by the local document test.

Inspired by your test i have done a extra test: open a document by the https URL directly with openoffice
`/Applications/LibreOffice.app/Contents/MacOS/soffice "https://apollo.virtorg.nl/alfresco/webdav/virtorg/documentLibrary/04_handleidingen/extern/VirtOrg_Solution_requirements.odt"`
It works well. LibreOffice is locking the document in Alfresco and updating is working as well. An extra proof that an URL document can be opened.

Of course opening with open opens the browser due to the https protocol

I will have a look how to report a bug to LibreOffice

Kind Regards

**Comment by gjongerh** [comment]
Hi, rather strange response. This command is also working. 
`/Applications/LibreOffice.app/Contents/MacOS/soffice "vnd.sun.star.webdavs://apollo.virtorg.nl/alfresco/webdav/virtorg/documentLibrary/04_handleidingen/extern/VirtOrg_Solution_requirements.odt"`

I have to go for a holiday. Futher testing after the hollidays

**Comment by SanjayBandhniya** [comment]
I am using libreoffice 6 with Alfresco 7 but not opening document.any other way to work?
